lib/home/balance/balance_filters_modal.dart

```diff
@@ -43,39 +57,35 @@ class _BalanceFiltersModalState extends State<BalanceFiltersModal> {
             ),
             Select(
               data: const [
-                SelectData(value: 'day', label: 'Dia'),
-                SelectData(value: 'week', label: 'Semana'),
-                SelectData(value: 'month', label: 'Mês'),
-                SelectData(value: 'year', label: 'Ano'),
+                SelectData(value: PeriodGroup.day, label: 'Dia'),
+                SelectData(value: PeriodGroup.week, label: 'Semana'),
+                SelectData(value: PeriodGroup.month, label: 'Mês'),
+                SelectData(value: PeriodGroup.year, label: 'Ano'),
               ],
               onPressed: (selected) {
-                setState(() {
-                  if (selectedValue == selected) {
-                    selectedValue = '';
-                    isButtonEnabled = false;
-                  } else {
-                    isButtonEnabled = true;
-                    selectedValue = selected;
-                  }
-                });
-              },
-              selectedValue: selectedValue,
-            ),
-            const SizedBox(height: 30),
-            DefaultButton(
-              isLoading: isButtonLoading,
-              enabled: isButtonEnabled,
-              onPressed: () {
-                try {
+                selected as PeriodGroup;
+                if (filtersCubit.state.periodGroup == selected) {
                   setState(() {
-                    isButtonLoading = true;
+                    buttonEnabled = false;
+                    state = state.copyWithPeriod(selected);
                   });
-                } finally {
+                } else {
                   setState(() {
-                    isButtonLoading = false;
+                    buttonEnabled = true;
+                    state = state.copyWithPeriod(selected);
                   });
                 }
```

Comment:
I can't suggest the change here, but this could be made differently:

```dart
setState(() {
   buttonEnabled = selected !=  filtersCubit.state.periodGroup;
   state = state.copyWithPeriod(selected);
});
```